### ISSUE TITLE
Webpacking ms-rest-js results in  TypeError: fetch is not a function

### ISSUE THREAD

**Issue by christiango** [Azure/ms-rest-js#373]
**Package Version**: 2.0.04
- [x] **nodejs**
   - **nodejs version**: 10.16.0
   - **os name/version**: Windows 10
- [ ] **browser**
    - **name/version**: ____
**Describe the bug**
We are consuming this library via `@azure/storage-blob`. For our scenario, we are running some code on our continuous integration pipeline that uploads files to blob storage. To avoid having to do an NPM install on the CI server, our desire is to webpack our file upload utility and just run the bundled JS file on the build server. However, we are seeing an issue resolving fetch that does not manifest when running the script in a non-bundled context.

**To Reproduce**
1. Create any script that consumes nodeFetchHttpClient.ts. In our case, we are bundling a modified version of the advanced example of the [@azure/storage-blob sdk](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-blob/samples/typescript/advanced.ts) with our blob storage details.
2. Create a webpack bundle with target "node".
3. Run that bundle with `node [bundlefilename].js`

**Expected behavior**
We expect the library to work in a bundled context, without having any issues relating to fetch.

**Screenshots**

**Additional context**
I dug a bit deeper here and I found that folks are having similar issues in other libraries, with the best example being the discussion here: https://github.com/bitinn/node-fetch/issues/450

It appears that in the case for ms-rest-jest, it is this line that is problematic for webpack:
```typescript
if (typeof globalWithFetch.fetch !== "function") {
  const fetch = require("node-fetch");
  globalWithFetch.fetch = fetch;
}
```

It appears to be an issue with commonJS vs ESM module resolution. It seems like webpack would want the import to be require("node-fetch").default. However, I am not sure if this would have implications for folks not consuming this library via webpack.

**Comment by mrsufgi** [comment]
I guess no one uses this repo, perhaps the js sdk repo will get better responses.

fyi we also have production issues with the same errors.  (typescript)

**Comment by hanvyj** [comment]
Any work-around for this issue? We're also facing it.

**Comment by daviwil** [comment]
This is now released in `@azure/ms-rest-js` 2.0.5.

**Comment by hanvyj** [comment]
Thanks @AnthonyMacKinnon working great now.
